From 4f8b725c971e7fb07bf19c6b969d12f29ff61f26 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:31 -0400
Subject: [tools] perf: Remove symbol_conf.use_callchain check

Message-id: <1408953210-26343-264-git-send-email-jolsa@redhat.com>
Patchwork-id: 89011
O-Subject: [PATCH RHEL7.1 BZ1133083 263/322] perf tools: Remove symbol_conf.use_callchain check
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 540476de74c9b11403656791838ede91405d3859
Author: Namhyung Kim <namhyung@kernel.org>
Date: Tue Jan 14 14:25:34 2014 +0900

description
===========
The machine__resolve_callchain() is called only if symbol_conf.
use_callchain is set so no need to check it again.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/machine.c b/tools/perf/util/machine.c
index 0130279..ded7459 100644
--- a/tools/perf/util/machine.c
+++ b/tools/perf/util/machine.c
@@ -1314,8 +1314,6 @@ static int machine__resolve_callchain_sample(struct machine *machine,
     *root_al = al;
     callchain_cursor_reset(&callchain_cursor);
    }
-   if (!symbol_conf.use_callchain)
-    break;
   }
 
   err = callchain_cursor_append(&callchain_cursor,
-- 
1.7.1